computation_graph/graph.py

```diff
@@ -123,3 +124,46 @@ def unbound_signature(
         kwargs=tuple(keep_not_in_bound_kwargs(node.signature.kwargs)),
         optional_kwargs=tuple(keep_not_in_bound_kwargs(node.signature.optional_kwargs)),
     )
+
+
+def replace_source(x, y):
+    return _transform_edges(_edge_source_equals(x), _replace_edge_source(y))
+
+
+def _transform_edges(query, edge_mapper):
+    return _operate_on_subgraph(
+        _split_by_condition(query), gamla.compose(tuple, gamla.map(edge_mapper))
+    )
+
+
+def _edge_source_equals(x):
+    if not isinstance(x, base_types.ComputationNode):
+        x = make_computation_node(x)
+    return lambda edge: edge.source == x
+
+
+def _replace_edge_source(replacement):
+    if not isinstance(replacement, base_types.ComputationNode):
+        replacement = make_computation_node(replacement)
+
+    def replace_edge_source(edge):
+        return dataclasses.replace(edge, source=replacement)
+
+    return replace_edge_source
```

Comment:
In `_edge_source_equals` we retuned a `lambda` and here we are returning an inner function. I think we need to be consistent here, @uriva Which do you prefer? 
I personally prefer the inner function pattern.

Reply:
probably an inner function or a composition